src/platform_impl/linux/wayland/window/state.rs

```diff
@@ -840,6 +848,35 @@ impl WindowState {
         }
     }
 
+    /// Make window background blurred
+    #[inline]
+    pub fn set_blur(&mut self, blurred: bool) {
+        if blurred && !self.is_blurred {
+            if self.blur.is_none() {
+                self.blur = self
+                    .blur_manager
+                    .as_ref()
+                    .map(|kbm| kbm.blur(self.window.wl_surface(), &self.queue_handle));
+            }
+            if let Some(blur) = &self.blur {
+                blur.commit();
+            } else {
+                info!("Blur manager unavailable, unable to change blur")
+            }
```

Comment:
This looks like `unwrap_or_else`.

Reply:
Not really, the else part doesn't give a KwinBlur instance